pkg/download/jsoncreator/json_creator.go

```diff
@@ -0,0 +1,115 @@
+package jsoncreator
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/api"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/rest"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/util"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/util/files"
+)
+
+//JSONCreator interface allows to mock the methods for unit testing
+type JSONCreator interface {
+	CreateJSONConfig(client rest.DynatraceClient, api api.Api, value api.Value, creator files.FileCreator,
+		path string) (name string, filter bool, err error)
+}
+
+//JSONCreatorImp object
+type JsonCreatorImp struct{}
+
+//NewJSONCreator creates a new instance of the jsonCreator
+func NewJSONCreator() *JsonCreatorImp {
+	result := JsonCreatorImp{}
+	return &result
+}
+
+//CreateJSONConfig creates a json file using the specified path and API data
+func (d *JsonCreatorImp) CreateJSONConfig(client rest.DynatraceClient, api api.Api, value api.Value, creator files.FileCreator,
+	path string) (name string, filter bool, err error) {
+	data, filter, err := getDetailFromAPI(client, api, value.Id)
+	if err != nil {
+		util.Log.Error("error getting detail %s from API", api.GetId())
+		return "", false, err
+	}
+	if filter == true {
+		return "", true, nil
+	}
+	jsonfile, err := processJSONFile(data, value.Id)
+
+	name, err = creator.CreateFile(jsonfile, path, value.Name, ".json")
+	if err != nil {
+		util.Log.Error("error writing detail %s", api.GetId())
+		return "", false, err
+	}
+	return name, false, nil
+}
+
+func getDetailFromAPI(client rest.DynatraceClient, api api.Api, name string) (dat map[string]interface{}, filter bool, err error) {
+
+	name = strings.Replace(name, "%", "%25", -1) // Replace the real % character in the name with %25 (encoded version)
```

Comment:
Do we only want to encode `%` or url encode the whole string? 

If we want the whole string url encoded, we could use `url.QueryEscape`.